crates/core/src/contracts/tests/account.rs

```diff
@@ -1,15 +1,53 @@
 #[cfg(test)]
 mod tests {
 
+    use std::sync::Arc;
+
     use kakarot_rpc_core::client::api::KakarotStarknetApi;
     use kakarot_test_utils::deploy_helpers::{
         get_contract, get_contract_deployed_bytecode, KakarotTestEnvironmentContext,
     };
     use kakarot_test_utils::fixtures::kakarot_test_env_ctx;
+    use reth_primitives::U256;
     use rstest::*;
     use starknet::core::types::{BlockId, BlockTag};
+    use starknet_crypto::FieldElement;
 
     use crate::contracts::account::{Account, KakarotAccount};
+    use crate::contracts::contract_account::ContractAccount;
+    use crate::mock::constants::ABDEL_STARKNET_ADDRESS;
+    use crate::mock::mock_starknet::{fixtures, mock_starknet_provider, AvailableFixtures};
+
+    #[tokio::test]
+    async fn test_nonce() {
+        // Given
+        let fixtures = fixtures(vec![AvailableFixtures::GetNonce]);
+        let starknet_provider = Arc::new(mock_starknet_provider(Some(fixtures)));
+        let contract_account = ContractAccount::new(*ABDEL_STARKNET_ADDRESS, &starknet_provider);
+
+        // When
+        let nonce = contract_account.nonce(&BlockId::Tag(BlockTag::Latest)).await.unwrap();
+
+        // Then
+        assert_eq!(U256::from(1), nonce);
+    }
+
+    #[tokio::test]
+    async fn test_implementation() {
+        // Given
+        let fixtures = fixtures(vec![AvailableFixtures::GetImplementation]);
+        let starknet_provider = Arc::new(mock_starknet_provider(Some(fixtures)));
+        let contract_account = ContractAccount::new(*ABDEL_STARKNET_ADDRESS, &starknet_provider);
+
+        // When
+        let implementation = contract_account.implementation(&BlockId::Tag(BlockTag::Latest)).await.unwrap();
+
+        // Then
+        assert_eq!(
+            FieldElement::from_hex_be("0x050de736bbc22be1cdb0c741afc80d01b6f5d1c248034a4cbe6f0eba6bde3051").unwrap(),
```

Comment:
how come this isnt "0x04ea3da1315fd04212a6f08f2ab7c9ef40cfb1cefa86633d9f9f308a1275367d";

Reply:
Fixed, the eoa and contract account class hashes got swapped